sdk/metricsadvisor/ai-metrics-advisor-rest/src/metricsAdvisorClient.ts

```diff
@@ -0,0 +1,144 @@
+import { ClientOptions } from "@azure-rest/core-client";
+import { isTokenCredential, TokenCredential } from "@azure/core-auth";
+import { OperationOptions } from "@azure/core-client";
+import { PagedAsyncIterableIterator, PageSettings } from "@azure/core-paging";
+import { GeneratedClientLike } from "./generated/clientDefinitions";
+import GeneratedClient from "./generated/generatedClient";
+import { AlertingResultQuery } from "./generated/models";
+import { AlertResultOutput } from "./generated/outputModels";
+import { AlertQueryTimeMode, AlertsPageResponse, AnomalyAlert } from "./models";
+import { paginate } from "./paginateHelper";
+import {
+  createMetricsAdvisorKeyCredentialPolicy,
+  MetricsAdvisorKeyCredential,
+  X_API_KEY_HEADER_NAME,
+} from "./metricsAdvisorKeyCredentialPolicy";
+/**
+ * Options for listing alerts
+ */
+export interface ListAlertsOptions extends OperationOptions {
+  /** Number of items to skip */
+  skip?: number;
+}
+
+export class MetricsAdvisorClient {
+  /**
+   * Url to service endpoint
+   */
+  public readonly endpointUrl: string;
+
+  /**
+   * A reference to the auto-generated MetricsAdvisor HTTP client.
+   */
+  private readonly client: GeneratedClientLike;
+
+  constructor(
+    endpointUrl: string,
+    credential: TokenCredential | MetricsAdvisorKeyCredential,
+    options: ClientOptions = {}
+  ) {
+    this.endpointUrl = endpointUrl;
+    this.client = GeneratedClient(endpointUrl, credential, {
+      ...options,
+      credentials: {
+        scopes: ["https://cognitiveservices.azure.com/.default"],
+        apiKeyHeaderName: X_API_KEY_HEADER_NAME
+      }
+    });
+    if (!isTokenCredential(credential)) {
+      this.client.pipeline.addPolicy(createMetricsAdvisorKeyCredentialPolicy(credential));
+    }
+  }
+
+  public listAlerts(
+    alertConfigId: string,
+    startTime: Date | string,
+    endTime: Date | string,
+    timeMode: AlertQueryTimeMode,
+    options: ListAlertsOptions = {}
+  ): PagedAsyncIterableIterator<AnomalyAlert, AlertsPageResponse> {
+    const iter = this.listItemsOfAlerts(
+      alertConfigId,
+      typeof startTime === "string" ? new Date(startTime) : startTime,
+      typeof endTime === "string" ? new Date(endTime) : endTime,
+      timeMode,
+      options
+    );
+    
+    return {
+      /**
+       * The next method, part of the iteration protocol
+       */
+      next() {
+        return iter.next();
+      },
+      /**
+       * The connection to the async iterator, part of the iteration protocol
+       */
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      /**
+       * @returns an AsyncIterableIterator that works a page at a time
+       */
+      // TODO: How to customise our pagination e.g feeding maxPageSize
```

Comment:
@joheredi  Here is the point that needs your help!

Reply:
I think we can set that as a request option when calling the initial response? If we can do the below we don't need to handle this in the generated code

```typescript
const initialResponse = client.path("/foo").get({queryParameters: {"$maxPageSize": 15}});
const iterator = paginate(client, initialResponse);
```

Reply:
We don't need to implement this functions for RLC, they should use the paginate helper as I described above. In RLC our strategy is different from other languages' DPG. Our goal is not to look like a High Level Client, but rather be close to REST and provide helper functions that our users may find useful, but are optional and decoupled from the client.

Reply:
Thanks for the explanation and updated from my side